Use existing button styles in include projects modal and sidebar

b12f opened the pull request:
The buttons in project include were using new styles, which caused
visual inconsistencies when compared to the rest of the application.
This commit changes those `spot-button`s to old `button`s. It also makes
sure that `spot-link` uses the theming variables, at least for the
non-danger variant.

Closes https://community.openproject.org/projects/openproject/work_packages/42251/activity

Kharonus committed:
[#42161] added new open file location link to api v3

dependabot[bot] committed:
Bump ejs from 3.1.6 to 3.1.7 in /frontend

Bumps [ejs](https://github.com/mde/ejs) from 3.1.6 to 3.1.7.
- [Release notes](https://github.com/mde/ejs/releases)
- [Changelog](https://github.com/mde/ejs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mde/ejs/compare/v3.1.6...v3.1.7)

---
updated-dependencies:
- dependency-name: ejs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Kharonus committed:
Merge branch 'dev' into implementation/42161-api-add-open-file-in-location-links-impl

Kharonus committed:
[#42161] fixed method call

Kharonus committed:
[#42161] fixed method call

Kharonus committed:
[#42173] Add endpoint (API spec) to update cache with live data

- https://community.openproject.org/work_packages/42173
- add new endpoint for updating file link data
- taken care of user endpoints and models for validation warnings and
  formatting

oliverguenther committed:
Add 12.1 release notes draft (#10604)

* Add 12.1 release notes draft

* Update Release notes for 12.1

Co-authored-by: birthe <b.lindenthal@openproject.com>

openprojectci committed:
update locales from crowdin [ci skip]

Commit:
Merge pull request #10601 from opf/implementation/42173-api-spec-add-endpoint-to-update-cache-with-live-data

[#42173] Add endpoint (API spec) to update cache with live data

Kharonus committed:
[#42204] add file links to work package resource

- https://community.openproject.org/work_packages/42204
- added file links include module for other resources
- fixed some rubocop issues in work package representer
- added representer tests

Kharonus committed:
[chore] bump ruby version to 3.1.2

ulferts committed:
Merge branch 'release/12.1' into dev

dombesz committed:
Fix installation doc config typo

Kharonus committed:
[chore] invalidated test cache key

Kharonus committed:
[chore] reverted ci image from bullseye to buster

Kharonus committed:
[chore] add net-pop

Kharonus committed:
[chore] add net-imap

openprojectci committed:
update locales from crowdin [ci skip]

dependabot[bot] committed:
Bump aws-sdk-s3 from 1.113.2 to 1.114.0

Bumps [aws-sdk-s3](https://github.com/aws/aws-sdk-ruby) from 1.113.2 to 1.114.0.
- [Release notes](https://github.com/aws/aws-sdk-ruby/releases)
- [Changelog](https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-s3/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-ruby/commits)

---
updated-dependencies:
- dependency-name: aws-sdk-s3
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

oliverguenther committed:
Merge pull request #10620 from opf/chore/bump-ruby-3-1-2

[chore] bump ruby version to 3.1.2

ulferts committed:
Merge branch 'release/12.1' into dev

Kharonus committed:
Merge branch 'dev' into implementation/42204-add-file-links-collection-to-work-package-resource

ulferts committed:
Merge branch 'release/12.1' into dev

openprojectci committed:
update locales from crowdin [ci skip]

dependabot[bot] committed:
Bump nokogiri from 1.13.4 to 1.13.5

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.13.4 to 1.13.5.
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.13.4...v1.13.5)

---
updated-dependencies:
- dependency-name: nokogiri
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ulferts committed:
Merge pull request #10637 from opf/dependabot/bundler/dev/nokogiri-1.13.5

Bump nokogiri from 1.13.4 to 1.13.5

ulferts committed:
Merge pull request #10573 from opf/dependabot/npm_and_yarn/frontend/ejs-3.1.7

Bump ejs from 3.1.6 to 3.1.7 in /frontend

marcalcobe committed:
Update README.md

cbliard committed:
trim trailing whitespace and insert final newline for ruby files

lindenthal committed:
Adding information about the current implementation (WIP)

lindenthal committed:
Merge pull request #10644 from opf/searchfield-addcases

Adding more cases for the search fields

openprojectci committed:
update locales from crowdin [ci skip]

dependabot[bot] committed:
Bump test-prof from 1.0.8 to 1.0.9

Bumps [test-prof](https://github.com/test-prof/test-prof) from 1.0.8 to 1.0.9.
- [Release notes](https://github.com/test-prof/test-prof/releases)
- [Changelog](https://github.com/test-prof/test-prof/blob/master/CHANGELOG.md)
- [Commits](https://github.com/test-prof/test-prof/compare/v1.0.8...v1.0.9)

---
updated-dependencies:
- dependency-name: test-prof
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ulferts committed:
Merge branch 'release/12.1' into dev

ulferts committed:
Merge branch 'release/12.1' into dev

ulferts committed:
Merge pull request #10646 from opf/dependabot/bundler/dev/test-prof-1.0.9

Bump test-prof from 1.0.8 to 1.0.9

openprojectci committed:
update locales from crowdin [ci skip]

dependabot[bot] committed:
Bump rubocop from 1.28.2 to 1.29.0

Bumps [rubocop](https://github.com/rubocop/rubocop) from 1.28.2 to 1.29.0.
- [Release notes](https://github.com/rubocop/rubocop/releases)
- [Changelog](https://github.com/rubocop/rubocop/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rubocop/rubocop/compare/v1.28.2...v1.29.0)

---
updated-dependencies:
- dependency-name: rubocop
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Kharonus committed:
[#42204] fixed tests and permission check

Kharonus committed:
Merge pull request #10571 from opf/implementation/42161-api-add-open-file-in-location-links-impl

[#42161] api add open file in location links implementation

Kharonus committed:
Merge pull request #10655 from opf/implementation/42204-add-file-links-collection-to-work-package-resource-short

[#42204] Add file links collection to work package resource (short)

birthe committed:
Update capitalization in BIM guide

cbliard committed:
make insert_final_newline = true for all files

lindenthal commented:
Hi @b12f,
thanks for working on this. The PP build is failing. Can you please check if this is related to this change?
Thanks
Niels

cbliard committed:
add matrix gem to get the Docker image to build again

matrix gem is no longer included in ruby 3.1 and is needed by the prawn gem, and prawn does not explicitly declare the dependency to the matrix gem in its latest version. We need to explicitly add it.

b12f commented:
@lindenthal it was related to a missing backend dependency, @cbliard was keen enough to notice and has pushed a fix. It should be available soon.

lindenthal commented:
hi @b12f,

I had a brief look at at the modal on iOS


![image](https://user-images.githubusercontent.com/1019287/167671492-6ac608ea-9f37-4e09-a6f4-60df091e9f8a.jpeg)

It looks different when compared with the user invite modal. 

![image](https://user-images.githubusercontent.com/1019287/167671950-f8928b75-9517-473d-9293-549330947a0c.jpeg)


Can you please check?



Thanks
Niels

b12f commented:
Hi @lindenthal this is intended. There are currently two patterns identified when it comes to modals:

1. context-dependent dropdowns (currently called `spot-drop-modal`) which the project-include modal is an instance of [figma link here](https://www.figma.com/file/XhCsrvs6rePifqbBpKYRWD/Components-Library?node-id=1408%3A6714), and
2. whole-screen modals (currently called `op-modal`). Even though there are already some ideas about how we might refactor them in the future, they have been left untouched. `spot-overlay-modal` has [some figma designs](https://www.figma.com/file/XhCsrvs6rePifqbBpKYRWD/Components-Library?node-id=625%3A4603), but there is nothing that has changed in code.

HDinger committed:
Merge branch 'release/12.1' into dev

marcalcobe commented:
Hi @b12f and @lindenthal,

After discussing this in the product team we've decided to only fix the distances between the buttons in the include projects modal as they are not correct. For the close/cancel action for the moment we will keep the implementations as they are as this are two different cases (full screen modal vs dropdown) and in the near future we will propose an aligned solution for mobile in this cases.

oliverguenther committed:
Merge remote-tracking branch 'origin/release/12.1' into dev

oliverguenther committed:
Explitly add matrix gem to gemfile for ruby 3.1

Commit:
Fix mobile button layout of project include

Kharonus committed:
[#42362] added storages link to project resource

Kharonus committed:
[#42362] added storages link to project representer

- added unit tests

oliverguenther committed:
Merge remote-tracking branch 'origin/release/12.1' into dev

openprojectci committed:
update locales from crowdin [ci skip]

Commit:
Merge pull request #10676 from opf/implementation/42362-api-add-storages-links-to-project-resource

[#42362] API: Add storages links to project resource

HDinger commented on frontend/src/app/spot/styles/sass/components/link.sass:
```diff
@@ -1,7 +1,7 @@
   @include unset-button-styles
   display: inline
-  color: $spot-color-main
+  color: var(--content-link-color)
```
This is not enough as it brings us back to the problem we originally wanted to solve with the new links. For the default theme (as well as the light theme by the way) the `content-link-hover-active-color` is the same as `content-link-color` resulting in no visible hover effect except the underlining. So with the theme colours we need the underlining as well.

marcalcobe replied:
Hi @b12f and @HDinger,

Sorry for the confusion on the mobile modal again! We where doing some trials with the Product Team on how we could harmonise the buttons in all the modals in the complete application. Please don't actualise anything in regards of this full width button on the mobile version. Also there is an error also solved where the clear selection button says cancel, please keep the clear selection option.

HDinger reviewed (changes requested):
I have some remarks:
* You forgot to replace the sidebar button
<img width="220" alt="Bildschirmfoto 2022-05-13 um 12 44 29" src="https://user-images.githubusercontent.com/7457313/168267751-493ce2ee-9b8d-46e2-a00b-aab6db1b395b.png">
 
* There is an issue with the link hover effect (see my comment).
* For mobile there is also still the new button visible and I think the margins don't match the specification.
<img width="150" alt="Bildschirmfoto 2022-05-13 um 12 47 18" src="https://user-images.githubusercontent.com/7457313/168268583-3b075f28-5b1a-4661-acbf-43de908e08e0.png">

**vs**
<img width="150" alt="Bildschirmfoto 2022-05-13 um 12 49 50" src="https://user-images.githubusercontent.com/7457313/168268530-f0b60b9a-951e-4dfc-ac73-5fce40893652.png">


* Please also have a look at the merge conflict in the Gemfile.

openprojectci committed:
update locales from crowdin [ci skip]

openprojectci committed:
update locales from crowdin [ci skip]

Commit:
Change sidebar buttons

Commit:
Add underline to spot-link hover

Commit:
Fix typo in gemfile

HDinger reviewed (changes requested):
I think you have accidentally overwritten parts of your older changes. For me it looks currently like this: 

<img width="1200" alt="Bildschirmfoto 2022-05-17 um 09 41 17" src="https://user-images.githubusercontent.com/7457313/168756700-b20c3fd4-6fac-4deb-8d2a-6e2f76284a4d.png">

The sidebar button is correct. However, the project include again uses the spot-buttons. Further, the link to add a new assignee has the wrong color (although the hover effect is correct).

HDinger commented:
One additional remark: This should be targeted agains the `release/12.1` branch in order to include it to the next patch release.

b12f commented:
Really weird, I did a rebase here and that apparently removed the commit that did the initial button changes. I'll see how I can fix this.

b12f commented:
Superseded by https://github.com/opf/openproject/pull/10692
